vision_datasets/dataset/vision_dataset.py

```diff
@@ -0,0 +1,176 @@
+import copy
+import logging
+import os.path
+import pathlib
+import typing
+
+from PIL import Image, JpegImagePlugin
+from tqdm import tqdm
+
+from ..common import DatasetTypes
+from ..data_manifest import DatasetManifest, ImageDataManifest, ImageLabelWithCategoryManifest
+from ..data_reader import FileReader, PILImageLoader
+from ..dataset.base_dataset import BaseDatasetInfo
+from .base_dataset import BaseDataset
+
+logger = logging.getLogger(__name__)
+
+
+class VisionDataset(BaseDataset):
+    """Dataset class that accesses data from dataset manifest
+    """
+
+    def __init__(self, dataset_info: BaseDatasetInfo, dataset_manifest: DatasetManifest, coordinates='relative', dataset_resources=None):
```

Comment:
coordinates is OD-specific, it's bit weird to have it here. We are not going to add all task-specific parameters here, right?

Reply:
that is a good pt, didn;t spend much time on the dataset class yet.

Let me add a to-do